src/style/makeTheme.ts

```diff
@@ -0,0 +1,135 @@
+import { defaultTheme } from './defaultTheme'
+import { Theme, ThemeBase } from './theme'
+import { cssUnit, parseColor, ThemeCSSRule } from './util'
+
+// Not declared in ./util since we don't want to export this function outside the package
+const handleCSSRule = (
+  rules: ThemeCSSRule | undefined,
+  defaultRule: ThemeCSSRule
+): ThemeCSSRule => {
+  if (!rules) return { css: defaultRule.css }
+  if (rules.replace)
+    return {
+      css: rules.css,
+    }
+  return {
+    css: (theme, props) => {
+      // Overwrites values from defaultStyle with the ones passed by ThemeCSSFunc
+      const styles = [defaultRule.css, rules.css]
+        .flat(2)
+        .map((st) => st(theme, props))
+        .flat() // TS Complains if we don't have this, even though we've used flat()
+      return styles
+    },
+  }
+}
+
+export const makeTheme = (theme?: Partial<ThemeBase<false>>): Theme => {
+  // This ternary operation allow us to avoid `TypeError: theme is undefined`
```

Comment:
makeTheme currently reduces the amount of work required to create a theme, for example most of our applications use a default light theme, instead of having us manually copying-pasting all the color values, then the radius, then all the rules for h1-h6, makeTheme will provide the default values in case we don't want to customize these.

Another feature `makeTheme` has is the ability to allow overwriting the default rules, for example we already have rules for h1-h6, but an application (say `virtual-coffee` for example) has all these in `color.primary.normal`. We can keep the default rules for h1-h6, and call makeTheme with:
```tsx
  const theme = makeTheme({
    font: {
      h1h6: {
        replace: false, // Keeps the original style, we don't want to clear it
        css: ({color}) => css`
          color: ${color.primary.normal};
        `,
      },
    },
  })
```

Note that in this case we didn't even declare the rules for colors, radius, buttons, inputs, etc. We wanted only to change the colors for headers. `makeTheme` in this example would call another helper function which would in turn merge the `defaultTheme` for `h1h6` with the new provided one.